### PR TITLE
ci: Move spellchecks to format_pre

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: ci: Move spellchecks to format_pre
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
im trying to understand why we have 2 sets of spellchecking

afaict also `spellcheck_pedantic fix` errors after the first incorrect spelling - and doesnt fix anything - so im wondering if there is any point to running the fix - ie just run the check

**Comment by phlax** [comment]
on the basis that i wasnt able to trigger the misspell spelling check, and that the fix doesnt work non-interactively for the pedantic spelling fix (as discussed offline) i have removed the first and set the latter to check only

cc @zuercher

File: ci/do_ci.sh [hunk 1]
```diff
@@ -419,22 +418,6 @@ elif [[ "$CI_TARGET" == "check_format" ]]; then
   "${ENVOY_SRCDIR}"/tools/code_format/check_format.py check
   BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS[*]}" "${ENVOY_SRCDIR}"/tools/proto_format/proto_format.sh check --test
   exit 0
-elif [[ "$CI_TARGET" == "check_spelling" ]]; then
```
Comment (alyssawilk) [inline ci/do_ci.sh:422]: hm, do you think we should keep these around for folks who are used to running things this way, or only want to do spelling not all pre checks?

We can also remove it now and add it back if anyone complains but I have these aliased so I'd mildly prefer to keep but I can totally update my aliases too :-)

File: ci/format_pre.sh [hunk 2]
```diff
@@ -58,6 +58,9 @@ bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/code_format:python_check -- --diff
 CURRENT=extensions
 bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/extensions:extensions_check
 
+CURRENT=spelling
+"${ENVOY_SRCDIR}"/tools/spelling/check_spelling_pedantic.py --mark check
```
Comment (alyssawilk) [inline ci/format_pre.sh:62]: why delete the other spell check?  I thought they caught different things?

File: ci/do_ci.sh (hunk 1 above)
Comment (phlax) [reply ci/do_ci.sh:422]: i think we need to remove these - i think the `do_ci.sh` switchboard is a bit of an anti pattern - the related readme has been updated

File: ci/format_pre.sh (hunk 2 above)
Comment (phlax) [reply ci/format_pre.sh:62]: afaict one is not catching anything

when this was discussed in the maintainer channel it was ~agreed to remove

Comment (alyssawilk) [reply ci/format_pre.sh:62]: ah I must have been out

**Review by alyssawilk — APPROVED** [review]

**Comment by RyanTheOptimist** [comment]
It looks like this PR broke some scripts (well shell aliases) I was using to check the spelling. (though I didn't notice immediately). Perhaps in the future it would be a good idea to announce workflow changes like this so developer know what they need to do?

**Pull request merged**